### PR TITLE
Call and retry AWS API

### LINKED ISSUE #45: AWS API Request limit exceeded exceptions
When API throttling is applied, the agent fails completely.

- Apply some retry
- Do not fail completely (?)

Might be related to #22

### PR DESCRIPTION
**Pull request by mohabusama**
+ Avoid API throttling exception

Attempt Fix #45

### CODE REVIEW

File: zmon_aws_agent/common.py [hunk 1]
```diff
@@ -1,5 +1,34 @@
...
+MAX_RETRIES = 10
+
+logger = logging.getLogger(__name__)
+
+
 def get_user_agent():
     return 'zmon-aws-agent/{}'.format(__version__)
+
+
+def call_and_retry(fn, *args, **kwargs):
```
Comment (hjacobs) [inline zmon_aws_agent/common.py]: maybe we can adapt this code to match what AWS suggests, see http://docs.aws.amazon.com/general/latest/gr/api-retries.html

Comment (mohabusama) [inline zmon_aws_agent/common.py]: Done. However, I increased the factor to 500 ms.

**Comment by hjacobs** [comment]
:+1:

**Comment by mohabusama** [comment]
👍

**Pull request merged**